zero/gkyl_gyrokinetic_maxwellian_correct.h

```diff
@@ -0,0 +1,93 @@
+  bool use_last_converged; // Boolean for if we are using the results of the iterative scheme
+                           // *even if* the scheme fails to converge. 
+  double mass; // Mass factor 
+  bool use_gpu; // bool for gpu useage
+  double eps; // tolerance for the iterator
+  int max_iter; // number of total iterations
+};
+
+// Correction status
+struct gkyl_gyrokinetic_maxwellian_correct_status {
+  bool iter_converged; // true if iterations converged
+  int num_iter; // number of iterations for the correction
+  double error[3]; // error in each moment (n, u_par, T/m)
+};  
+
+/**
+ * Create new updater to correct the gyrokinetic Maxwellian distribution function
+ * so that its moments match desired input moments.
+ *
+ * @param inp Input parameters defined in gkyl_gyrokinetic_maxwellian_correct_inp struct.
+ * @return New updater pointer.
+ */
+struct gkyl_gyrokinetic_maxwellian_correct* 
+gkyl_gyrokinetic_maxwellian_correct_inew(const struct gkyl_gyrokinetic_maxwellian_correct_inp *inp);
+
+/**
+ * Fix the gyrokinetic Maxwellian distribution function so that its density matches a target density.
+ *
+ * @param up gyrokinetic Maxwellian distribution function moment correction updater
+ * @param f_max gyrokinetic Maxwellian distribution function to fix (modified in-place)
+ * @param moms_target Target density, n
+ * @param phase_local Local phase-space range
+ * @param conf_local Local configuration space range
+ */
+void
+gkyl_gyrokinetic_maxwellian_correct_density_moment(gkyl_gyrokinetic_maxwellian_correct *up,
+  struct gkyl_array *f_max, const struct gkyl_array *moms_target, 
+  const struct gkyl_range *phase_local, const struct gkyl_range *conf_local);
+
+/**
+ * Fix the gyrokinetic Maxwellian distribution function
+ * so that *all* its primitive moments (n, u_par, T/m) match target moments.
+ * NOTE: If this algorithm fails, the returns the original distribution function
+ * with only the desired density moment corrected.
+ *
+ * @param up gyrokinetic Maxwellian distribution function moment correction updater
+ * @param f_max gyrokinetic Maxwellian distribution function to fix (modified in-place)
+ * @param moms_target Target primitive moments (n, u_par, T/m)
+ * @param phase_local Local phase-space range
+ * @param conf_local Local configuration space range
+ * @return Status of correction
+ */
+struct gkyl_gyrokinetic_maxwellian_correct_status 
+gkyl_gyrokinetic_maxwellian_correct_all_moments(gkyl_gyrokinetic_maxwellian_correct *up,
+  struct gkyl_array *f_max, const struct gkyl_array *moms_target, 
+  const struct gkyl_range *phase_local, const struct gkyl_range *conf_local);
+
+/**
+ * Host-side wrapper for computing the absolute value of the 
+ * difference in cell averages between the target moments and iterative moments.
+ */
+void gkyl_gyrokinetic_maxwellian_correct_all_moments_abs_diff_cu(const struct gkyl_range *conf_range, 
```

Comment:
@JunoRavin can you put this declaration in the private header? also, please describe its arguments.